src/ClusterManager/joblog_manager.py

```diff
@@ -31,11 +31,14 @@
 from osUtils import mkdirsAsUser
 from config import config, GetStoragePath
 from DataHandler import DataHandler
+from JobLogUtils import GetJobLog
 
 from cluster_manager import setup_exporter_thread, manager_iteration_histogram, register_stack_trace_dump, update_file_modification_time, record
 
 logger = logging.getLogger(__name__)
 
+elasticsearch_deployed = len(config.get('elasticsearch', [])) > 0
```

Comment:
If `elasticsearch` is empty in `config.yaml`, `config.get('elasticsearch', [])` will give `None`.

Reply:
Did you mean explicitly set
```yaml
elasticsearch:
```
?

Reply:
Upon absence of elasticsearch nodes, it's 
```
elasticsearch:
```
in `config.yaml`. Then `config.get('elasticsearch', [])` gives constant `None`, because key `elasticsearch` exists and its value is `None`. `len(None)` will crash.
I am suggesting something like
```
elasticsearch_deployed = config.get('elasticsearch', []) is not None and len(config.get('elasticsearch', [])) > 0
```
```

Reply:
Fixed as `config.get('elasticsearch') is list and len(config['elasticsearch']) > 0`